craft_cli/dispatcher.py

```diff
@@ -281,7 +281,7 @@ def _build_usage_exc(self, text: str) -> ArgumentParsingError:
         return ArgumentParsingError(self._help_builder.get_usage_message(text))
 
     def _get_requested_help(  # noqa: PLR0912 (too many branches)
```

Comment:
suggestion: The signature changed and giving a not None value is only indirectly tested (while testing `dispatcher.pre_parse_args()`). 

Maybe add at least one test in `test_help.py` to make sure `_get_requested_help()` is behaving as expected with some values (especially not a dict, that will make the command unhappy)?

Reply:
I added a new test that explicitly doesn't pass the second parameter to `pre_parse_args()`, thanks for pointing this out.

In my opinion this module already tests too much of the internals. It's a problem in `craft-cli` as a whole so now I'm trying to stick to the public interfaces and the expected, user-visible outcomes.